tools/distrib/build_ruby_environment_macos.sh

```diff
@@ -51,15 +51,28 @@ EOF
 
 MAKE="make -j8"
 
-# Install ruby 3.0.0 for rake-compiler
-# Download ruby 3.0.0 sources outside of the cross-ruby.rake file, since the
+
+# Download ruby 3.0.0 and 3.1.0 sources outside of the cross-ruby.rake file, since the
 # latest rake-compiler/v1.1.1 cross-ruby.rake file requires tar.bz2 source
 # files.
 # TODO(apolcyn): remove this hack when tar.bz2 sources are available for ruby
 # 3.0.0 in https://ftp.ruby-lang.org/pub/ruby/3.0/. Also see
 # https://stackoverflow.com/questions/65477613/rvm-where-is-ruby-3-0-0.
 set +x # rvm commands are very verbose
 source ~/.rvm/scripts/rvm
+# Install ruby 3.1.0 for rake-compiler
+echo "rvm use 3.1.0"
+rvm use 3.1.0
+set -x
+RUBY_3_1_0_TAR="${HOME}/.rake-compiler/sources/ruby-3.1.0.tar.gz"
+mkdir -p "$(dirname $RUBY_3_1_0_TAR)"
+curl -L "https://ftp.ruby-lang.org/pub/ruby/3.0/$(basename $RUBY_3_1_0_TAR)" -o "$RUBY_3_1_0_TAR"
```

Comment:
this 3.0 needs to be changed to 3.1,  (current url doesn't exist)

Reply:
I updated it and started the tests again.